packages/bpk-docs/src/components/DocsPageWrapper/DocsPageWrapper.js

```diff
@@ -34,106 +31,146 @@ import STYLES from './DocsPageWrapper.scss';
   >
+    <BpkHorizontalNavItem
+      name="android"
+      disabled={disableAndroidTab}
+      selected={platform === 'android'}
+      onClick={onAndroidClick}
+    >
+      Android
+    </BpkHorizontalNavItem>
+    <BpkHorizontalNavItem
+      name="ios"
+      disabled={disableIOSTab}
+      selected={platform === 'ios'}
+      onClick={onIOSClick}
+    >
+      iOS
+    </BpkHorizontalNavItem>
     <BpkHorizontalNavItem
       name="native"
       disabled={disableNativeTab}
       selected={platform === 'native'}
       onClick={onNativeClick}
     >
-      <AlignedBpkSmallMobileIcon
-        className={getClassName('bpkdocs-page-wrapper__platform-icon')}
-      />
-      Native
+      React Native
     </BpkHorizontalNavItem>
     <BpkHorizontalNavItem
       name="web"
       disabled={disableWebTab}
       selected={platform === 'web'}
       onClick={onWebClick}
     >
-      <AlignedBpkSmallWindowIcon
-        className={getClassName('bpkdocs-page-wrapper__platform-icon')}
-      />
       Web
     </BpkHorizontalNavItem>
   </BpkHorizontalNav>
 );
 
 PlatformNav.propTypes = {
-  platform: PropTypes.oneOf(['web', 'native']).isRequired,
+  platform: PropTypes.oneOf(['android', 'ios', 'native', 'web']).isRequired,
+  onAndroidClick: PropTypes.func.isRequired,
+  onIOSClick: PropTypes.func.isRequired,
   onNativeClick: PropTypes.func.isRequired,
   onWebClick: PropTypes.func.isRequired,
+  disableAndroidTab: PropTypes.bool.isRequired,
+  disableIOSTab: PropTypes.bool.isRequired,
   disableNativeTab: PropTypes.bool.isRequired,
   disableWebTab: PropTypes.bool.isRequired,
 };
 
 const DocsPageWrapper = props => {
   const {
     blurb,
+    androidSubpage,
+    iosSubpage,
     nativeSubpage,
-    title,
     webSubpage,
+    title,
     match,
     history,
     location,
   } = props;
   const path = match.url;
-  let renderWeb = location.search.indexOf('platform=web') >= 0;
-  let renderNative = location.search.indexOf('platform=native') >= 0;
 
-  if (!renderWeb && !renderNative) {
-    renderNative = !!nativeSubpage;
-    renderWeb = !!webSubpage && !renderNative;
-    history.replace(`${path}?platform=${renderNative ? 'native' : 'web'}`);
+  const platforms = {
+    android: androidSubpage,
+    ios: iosSubpage,
+    native: nativeSubpage,
+    web: webSubpage,
+  };
+
+  let initiallySelectedPlatform = 'web';
+  let initiallyRenderedSubpage = webSubpage;
+  if (androidSubpage) {
+    initiallySelectedPlatform = 'android';
+    initiallyRenderedSubpage = androidSubpage;
+  } else if (iosSubpage) {
+    initiallySelectedPlatform = 'ios';
+    initiallyRenderedSubpage = iosSubpage;
+  } else if (nativeSubpage) {
+    initiallySelectedPlatform = 'native';
+    initiallyRenderedSubpage = nativeSubpage;
+  }
+
+  const platformQueryParamMatches = platformQueryParamRegex.exec(
+    location.search,
+  );
+  if (
+    platformQueryParamMatches &&
+    platformQueryParamMatches[1] &&
```

Comment:
Is this necessary? The group is not optional so if the regex matches it should be always there. Also instead of verifying if the platform name is valid here you could add it to the regex. _`/platform=(android|ios|native|web)/`; and make the regex case insensitive

Reply:
Good idea, I'll do that